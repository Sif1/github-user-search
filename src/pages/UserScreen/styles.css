.user-wrapper {
    text-align: center;
    background-color: #628283;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .user-title {
      font-size: calc(30px + 2vmin);
      color: white;
  }
  .description-text {
      font-size: calc(20px);
      color: #b3f0ce;
  }

  .back-button {
      width: 180px;
      margin: 10px;
      padding-left: 6px;
      padding-right: 6px;
      border-radius: 8px;
  }

  .error-message {
    font-size: calc(20px);
    color: #b3f0ce;
  }

  .img-style {
      height: 260px;
      width: 260px;
  }

  .user-section {
    align-items: center;
  }

  .repo-section {
    margin-left: 200px;
    margin-right: 200px;
  }

  .repo-name {
    font-size: 24px;
    color: #021a0d;
  }

  .repo-description {
    font-size: 16px;
    padding-bottom: 30px;
    border-bottom: solid 1px rgb(73, 224, 53);
  }