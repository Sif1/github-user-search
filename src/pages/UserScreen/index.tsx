import React, { useState, useEffect, useCallback } from 'react';
import {
    useParams,
    useNavigate,
  } from "react-router-dom";
import './styles.css';

function UserScreen() {

    const [userInfo, setUserInfo] = useState<any>();
    const [userRepos, setUserRepos] = useState<any>();
    const [shouldShowError, setShouldShowError] = useState(false);

    let userName = useParams();
    const navigate = useNavigate();

    const handleFetchPublicRepos = useCallback(() => {
            fetch(`https://api.github.com/users/${userName.id}/repos`)
            .then(response => response.json())
            .then(data => {
                setUserRepos(data)});
    }, [userName.id]);

    const handleGoBack = () => {
        navigate(-1);
    }
    
    useEffect(() => {
        fetch(`https://api.github.com/users/${userName.id}`)
        .then(response => response.json())
        .then(data => {
            if (!data.login) {
                setShouldShowError(true);
            }
            setUserInfo(data)
            handleFetchPublicRepos();
        });
      }, [handleFetchPublicRepos, userName.id]);

    

  return (
    <>
        <div className="user-wrapper">
            <h3 className="user-title">
            {`Here's some info about the user: ${userName.id}`}
            </h3>
            <button onClick={handleGoBack} className='back-button'>
                Search another user
            </button>
            <div className='user-section'>
                {!shouldShowError ? 
                <>
                    <img 
                        alt='user avatar'
                        src={userInfo?.avatar_url}
                        className='img-style'
                    />
                    <p className='description-text'>
                        {userName.id} has {userInfo?.public_repos} public repositories on his github account. See more about them below!
                    </p>
                    {userRepos?.map((repo: any, i: number) => (
                        <div key={i} className='repo-section'>
                            <p className='repo-name'>
                                Repo {++i}: {repo.name}
                            </p>
                            <p className='repo-description'>
                                Description: {repo.description}
                            </p>
                        </div>
                    ))}
                </> : <p className='error-message'>user {userName.id} not found </p>
                 
                }
            </div>
        </div>
        
    </>
  );
}

export default UserScreen;
