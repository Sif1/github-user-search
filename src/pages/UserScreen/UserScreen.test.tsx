/* eslint-disable import/first */
const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
   useNavigate: () => mockedUsedNavigate,
 }));

import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import React from 'react';
import renderer from 'react-test-renderer';
import UserScreen from './';

test('renders userscreen correctly - snapshot', () => {
  const tree = renderer
    .create(<UserScreen/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('user screen render title', () => {
    render(<UserScreen />);
    const titleText = screen.getByText(/Here's some info about the user:/i);
    expect(titleText).toBeInTheDocument();
  });

test('navigate back button should be enabled on first render', () => {
    render(<UserScreen />);
    const searchButton = screen.getByRole('button');
    expect(searchButton).toBeEnabled();
  });