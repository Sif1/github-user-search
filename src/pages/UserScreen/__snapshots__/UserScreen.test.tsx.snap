// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders userscreen correctly - snapshot 1`] = `
<div
  className="user-wrapper"
>
  <h3
    className="user-title"
  >
    Here's some info about the user: undefined
  </h3>
  <button
    className="back-button"
    onClick={[Function]}
  >
    Search another user
  </button>
  <div
    className="user-section"
  >
    <img
      className="img-style"
    />
    <p
      className="description-text"
    >
       has 
       public repositories on his github account. See more about them below!
    </p>
  </div>
</div>
`;
